lib/css-converter.js

```diff
@@ -0,0 +1,319 @@
+    case '*=':
+      if (['description', 'text'].includes(attrName)) {
+        return `.${methodName}Contains("${value}")`;
+      }
+      return `.${methodName}Matches("${_.escapeRegExp(value)}")`;
+    case '^=':
+      if (['description', 'text'].includes(attrName)) {
+        return `.${methodName}StartsWith("${value}")`;
+      }
+      return `.${methodName}Matches("^${_.escapeRegExp(value)}")`;
+    case '$=':
+      return `.${methodName}Matches("${_.escapeRegExp(value)}$")`;
+    case '~=':
+      return `.${methodName}Matches("${getWordMatcherRegex(value)}")`;
+    default:
+      // Unreachable, but adding error in case a new CSS attribute is added.
+      throw new Error(`Unsupported CSS attribute operator '${cssAttr.operator}'. ` +
+        ` '=', '*=', '^=', '$=' and '~=' are supported.`);
+  }
+}
+
+/**
+ * @typedef {Object} CssPseudo
+ * @property {?string} valueType The type of CSS pseudo selector (https://www.npmjs.com/package/css-selector-parser for reference)
+ * @property {?string} name The name of the pseudo selector
+ * @property {?string} value The value of the pseudo selector
+ */
+
+/**
+ * Convert a CSS pseudo class to a UiSelector
+ *
+ * @param {CssPseudo} cssPseudo CSS Pseudo class
+ * @returns {string} Pseudo selector parsed as UiSelector
+ */
+function parsePseudo (cssPseudo) {
+  if (cssPseudo.valueType && cssPseudo.valueType !== 'string') {
+    throw new Error(`'${cssPseudo.name}=${cssPseudo.value}'. ` +
+      `Unsupported css pseudo class value type: '${cssPseudo.valueType}'. Only 'string' type or empty is supported.`);
+  }
+
+  const pseudoName = assertGetAttrName(cssPseudo);
+
+  if (BOOLEAN_ATTRS.includes(pseudoName)) {
+    return `.${toSnakeCase(pseudoName)}(${assertGetBool(cssPseudo)})`;
+  }
+
+  if (NUMERIC_ATTRS.includes(pseudoName)) {
+    return `.${pseudoName}(${cssPseudo.value})`;
+  }
+}
+
+/**
+ * @typedef {Object} CssRule
+ * @property {?string} nestingOperator The nesting operator (aka: combinator https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors)
+ * @property {?string} tagName The tag name (aka: type selector https://developer.mozilla.org/en-US/docs/Web/CSS/Type_selectors)
+ * @property {?string[]} classNames An array of CSS class names
+ * @property {?CssAttr[]} attrs An array of CSS attributes
+ * @property {?CssPseudo[]} attrs An array of CSS pseudos
+ * @property {?string} id CSS identifier
+ * @property {?CssRule} rule A descendant of this CSS rule
+ */
+
+/**
+ * Convert a CSS rule to a UiSelector
+ * @param {CssRule} cssRule CSS rule definition
+ */
+function parseCssRule (cssRule) {
+  const { nestingOperator } = cssRule;
+  if (nestingOperator && nestingOperator !== ' ') {
+    throw new Error(`'${nestingOperator}' is not a supported combinator. ` +
+      `Only child combinator (>) and descendant combinator are supported.`);
+  }
+
+  let uiAutomatorSelector = 'new UiSelector()';
+  if (cssRule.tagName && cssRule.tagName !== '*') {
```

Comment:
Not sure how to do this differently, but I always have some issues reading multiple `if|else`-statements like this